app/cdap/components/shared/RichToolTip/RichToolTip.tsx

```diff
@@ -0,0 +1,159 @@
+import {
+  Box,
+  ClickAwayListener,
+  Fade,
+  makeStyles,
+  Paper,
+  Popper,
+  PopperPlacementType,
+} from '@material-ui/core';
+import React, { useState, ReactElement, cloneElement } from 'react';
+
+interface IProps {
+  content: ReactElement;
+  children: ReactElement;
+  open: boolean;
+  onClose?: () => void;
+  arrow?: boolean;
+  placement?: PopperPlacementType;
+}
+
+const useStyles = makeStyles((theme) => {
+  const color = theme.palette.background.paper;
+  return {
+    popoverRoot: {
+      backgroundColor: color,
+      maxWidth: 1000,
+    },
+    content: {
+      padding: theme.spacing(2),
+    },
+    popper: {
+      zIndex: 2000,
+      '&[x-placement*="bottom"] $arrow': {
+        top: 0,
+        left: 0,
+        marginTop: '-0.71em',
+        marginLeft: 4,
+        marginRight: 4,
+        '&::before': {
+          transformOrigin: '0 100%',
+        },
+      },
+      '&[x-placement*="top"] $arrow': {
+        bottom: 0,
+        left: 0,
+        marginBottom: '-0.71em',
+        marginLeft: 4,
+        marginRight: 4,
+        '&::before': {
+          transformOrigin: '100% 0',
+        },
+      },
+      '&[x-placement*="right"] $arrow': {
+        left: 0,
+        marginLeft: '-0.71em',
+        height: '1em',
+        width: '0.71em',
+        marginTop: 4,
+        marginBottom: 4,
+        '&::before': {
+          transformOrigin: '100% 100%',
+        },
+      },
+      '&[x-placement*="left"] $arrow': {
+        right: 0,
+        marginRight: '-0.71em',
+        height: '1em',
+        width: '0.71em',
+        marginTop: 4,
+        marginBottom: 4,
+        '&::before': {
+          transformOrigin: '0 0',
+        },
+      },
+    },
+    arrow: {
+      overflow: 'hidden',
+      position: 'absolute',
+      width: '1em',
+      height: '0.71em' /* = width / sqrt(2) = (length of the hypotenuse) */,
+      boxSizing: 'border-box',
+      color,
+      '&::before': {
+        content: '""',
+        margin: 'auto',
+        display: 'block',
+        width: '100%',
+        height: '100%',
+        boxShadow: theme.shadows[1],
+        backgroundColor: 'currentColor',
+        transform: 'rotate(45deg)',
+      },
+    },
+  };
+});
+
+/**
+ * Rich tooltip is a component taken from this stackoverflow answer
+ * https://stackoverflow.com/questions/57500324/example-of-using-arrow-in-popper-component-of-material-ui
+ * Once we upgrade material ui we can use the regular tooltip more easily - the arrow portion of the popper
+ * was very difficult to use.
+ */
+
+const RichTooltip = ({
```

Comment:
Should this file be named `index.tsx`?

Reply:
This is something i feel weird about - on one hand it makes it uniform - on the other it makes everything named index.tsx or ts so its hard to recognize the files quickly like in the code tabs of vs code. do you have any strong opinions?  if its a one file component it should probably not be in its own folder and then add a folder if theres multiple files?

Reply:
I also think we should stick to one or the other. i.e., using index for the main file or just the component (folder) name. Don't feel strongly about either options as long as we're consistent.

Reply:
I've been doing Sean's way, since he told me so

Reply:
nit: update file name to `index.tsx`